action/protocol/staking/protocol.go

```diff
@@ -182,8 +182,13 @@ func (p *Protocol) Start(ctx context.Context) error {
 	return nil
 }
 
-// CreatePreStates creates states before Handle
-func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager) error {
+// ActionPostHandler performs post-handler action
+func (p *Protocol) ActionPostHandler(ctx context.Context, sm protocol.StateManager) error {
+	return sm.Load(protocolID, p.inMemCandidates)
+}
+
+// ActionPostCommit performs post-commit action
+func (p *Protocol) ActionPostCommit(ctx context.Context, sm protocol.StateManager) error {
```

Comment:
Upon factory.Commit() success, this is called to update the CandidateCenter, to prepare it for next time

Reply:
so that we only update when necessary, instead of update for each block in CreatePreStates()